fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/HNSW.java

```diff
@@ -0,0 +1,2022 @@
+        if (logger.isTraceEnabled()) {
+            logger.trace("new node with key={} selected to be inserted into layer={}", newPrimaryKey, insertionLayer);
+        }
+
+        return StorageAdapter.fetchAccessInfo(getConfig(), transaction, getSubspace(), getOnReadListener())
+                .thenCompose(accessInfo -> {
+                    final AccessInfo currentAccessInfo;
+                    final AffineOperator storageTransform = storageTransform(accessInfo);
+                    final RealVector transformedNewVector = storageTransform.applyInvert(newVector);
+                    final Quantizer quantizer = quantizer(accessInfo);
+                    final Estimator estimator = quantizer.estimator();
+
+                    if (accessInfo == null) {
+                        // this is the first node
+                        writeLonelyNodes(quantizer, transaction, newPrimaryKey, transformedNewVector,
+                                insertionLayer, -1);
+                        currentAccessInfo = new AccessInfo(
+                                new EntryNodeReference(newPrimaryKey, transformedNewVector, insertionLayer),
+                                -1L, null);
+                        StorageAdapter.writeAccessInfo(transaction, getSubspace(), currentAccessInfo,
+                                getOnWriteListener());
+                        if (logger.isTraceEnabled()) {
+                            logger.trace("written initial entry node reference with key={} on layer={}",
+                                    newPrimaryKey, insertionLayer);
+                        }
+                    } else {
+                        final EntryNodeReference entryNodeReference = accessInfo.getEntryNodeReference();
+                        final int lMax = entryNodeReference.getLayer();
+                        if (insertionLayer > lMax) {
+                            writeLonelyNodes(quantizer, transaction, newPrimaryKey, transformedNewVector,
+                                    insertionLayer, lMax);
+                            currentAccessInfo = accessInfo.withNewEntryNodeReference(
+                                    new EntryNodeReference(newPrimaryKey, transformedNewVector,
+                                            insertionLayer));
+                            StorageAdapter.writeAccessInfo(transaction, getSubspace(), currentAccessInfo,
+                                    getOnWriteListener());
+                            if (logger.isTraceEnabled()) {
+                                logger.trace("written higher entry node reference with key={} on layer={}",
+                                        newPrimaryKey, insertionLayer);
+                            }
+                        } else {
+                            currentAccessInfo = accessInfo;
+                        }
+                    }
+
+                    if (accessInfo == null) {
+                        return AsyncUtil.DONE;
```

Comment:
This should only be hit if the `accessInfo` was originally `null`, right? So, logically, this would be the same as returning `AsyncUtil.DONE` within the `if` block that inserts the first (and only) node into the tree, and then this would just assert that `accessInfo` is not `null`

Reply:
changed this to just return in the earlier `if`